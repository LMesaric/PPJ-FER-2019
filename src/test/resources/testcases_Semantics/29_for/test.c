int main(void) {
    int i, y=0;
    for (i=0; i<10; ++i) {
        i = i + 1;
        y = y + 2;
    }
    for (i=0; main; ++i) {
        y = 1;
    }
    return y;
}
