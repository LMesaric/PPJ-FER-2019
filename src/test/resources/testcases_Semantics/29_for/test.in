<prijevodna_jedinica>
 <vanjska_deklaracija>
  <definicija_funkcije>
   <ime_tipa>
    <specifikator_tipa>
     KR_INT 1 int
   IDN 1 main
   L_ZAGRADA 1 (
   KR_VOID 1 void
   D_ZAGRADA 1 )
   <slozena_naredba>
    L_VIT_ZAGRADA 1 {
    <lista_deklaracija>
     <deklaracija>
      <ime_tipa>
       <specifikator_tipa>
        KR_INT 2 int
      <lista_init_deklaratora>
       <lista_init_deklaratora>
        <init_deklarator>
         <izravni_deklarator>
          IDN 2 i
       ZAREZ 2 ,
       <init_deklarator>
        <izravni_deklarator>
         IDN 2 y
        OP_PRIDRUZI 2 =
        <inicijalizator>
         <izraz_pridruzivanja>
          <log_ili_izraz>
           <log_i_izraz>
            <bin_ili_izraz>
             <bin_xili_izraz>
              <bin_i_izraz>
               <jednakosni_izraz>
                <odnosni_izraz>
                 <aditivni_izraz>
                  <multiplikativni_izraz>
                   <cast_izraz>
                    <unarni_izraz>
                     <postfiks_izraz>
                      <primarni_izraz>
                       BROJ 2 0
      TOCKAZAREZ 2 ;
    <lista_naredbi>
     <lista_naredbi>
      <lista_naredbi>
       <naredba>
        <naredba_petlje>
         KR_FOR 3 for
         L_ZAGRADA 3 (
         <izraz_naredba>
          <izraz>
           <izraz_pridruzivanja>
            <postfiks_izraz>
             <primarni_izraz>
              IDN 3 i
            OP_PRIDRUZI 3 =
            <izraz_pridruzivanja>
             <log_ili_izraz>
              <log_i_izraz>
               <bin_ili_izraz>
                <bin_xili_izraz>
                 <bin_i_izraz>
                  <jednakosni_izraz>
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          BROJ 3 0
          TOCKAZAREZ 3 ;
         <izraz_naredba>
          <izraz>
           <izraz_pridruzivanja>
            <log_ili_izraz>
             <log_i_izraz>
              <bin_ili_izraz>
               <bin_xili_izraz>
                <bin_i_izraz>
                 <jednakosni_izraz>
                  <odnosni_izraz>
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          IDN 3 i
                   OP_LT 3 <
                   <aditivni_izraz>
                    <multiplikativni_izraz>
                     <cast_izraz>
                      <unarni_izraz>
                       <postfiks_izraz>
                        <primarni_izraz>
                         BROJ 3 10
          TOCKAZAREZ 3 ;
         <izraz>
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      OP_INC 3 ++
                      <unarni_izraz>
                       <postfiks_izraz>
                        <primarni_izraz>
                         IDN 3 i
         D_ZAGRADA 3 )
         <naredba>
          <slozena_naredba>
           L_VIT_ZAGRADA 3 {
           <lista_naredbi>
            <lista_naredbi>
             <naredba>
              <izraz_naredba>
               <izraz>
                <izraz_pridruzivanja>
                 <postfiks_izraz>
                  <primarni_izraz>
                   IDN 4 i
                 OP_PRIDRUZI 4 =
                 <izraz_pridruzivanja>
                  <log_ili_izraz>
                   <log_i_izraz>
                    <bin_ili_izraz>
                     <bin_xili_izraz>
                      <bin_i_izraz>
                       <jednakosni_izraz>
                        <odnosni_izraz>
                         <aditivni_izraz>
                          <aditivni_izraz>
                           <multiplikativni_izraz>
                            <cast_izraz>
                             <unarni_izraz>
                              <postfiks_izraz>
                               <primarni_izraz>
                                IDN 4 i
                          PLUS 4 +
                          <multiplikativni_izraz>
                           <cast_izraz>
                            <unarni_izraz>
                             <postfiks_izraz>
                              <primarni_izraz>
                               BROJ 4 1
               TOCKAZAREZ 4 ;
            <naredba>
             <izraz_naredba>
              <izraz>
               <izraz_pridruzivanja>
                <postfiks_izraz>
                 <primarni_izraz>
                  IDN 5 y
                OP_PRIDRUZI 5 =
                <izraz_pridruzivanja>
                 <log_ili_izraz>
                  <log_i_izraz>
                   <bin_ili_izraz>
                    <bin_xili_izraz>
                     <bin_i_izraz>
                      <jednakosni_izraz>
                       <odnosni_izraz>
                        <aditivni_izraz>
                         <aditivni_izraz>
                          <multiplikativni_izraz>
                           <cast_izraz>
                            <unarni_izraz>
                             <postfiks_izraz>
                              <primarni_izraz>
                               IDN 5 y
                         PLUS 5 +
                         <multiplikativni_izraz>
                          <cast_izraz>
                           <unarni_izraz>
                            <postfiks_izraz>
                             <primarni_izraz>
                              BROJ 5 2
              TOCKAZAREZ 5 ;
           D_VIT_ZAGRADA 6 }
      <naredba>
       <naredba_petlje>
        KR_FOR 7 for
        L_ZAGRADA 7 (
        <izraz_naredba>
         <izraz>
          <izraz_pridruzivanja>
           <postfiks_izraz>
            <primarni_izraz>
             IDN 7 i
           OP_PRIDRUZI 7 =
           <izraz_pridruzivanja>
            <log_ili_izraz>
             <log_i_izraz>
              <bin_ili_izraz>
               <bin_xili_izraz>
                <bin_i_izraz>
                 <jednakosni_izraz>
                  <odnosni_izraz>
                   <aditivni_izraz>
                    <multiplikativni_izraz>
                     <cast_izraz>
                      <unarni_izraz>
                       <postfiks_izraz>
                        <primarni_izraz>
                         BROJ 7 0
         TOCKAZAREZ 7 ;
        <izraz_naredba>
         <izraz>
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        IDN 7 main
         TOCKAZAREZ 7 ;
        <izraz>
         <izraz_pridruzivanja>
          <log_ili_izraz>
           <log_i_izraz>
            <bin_ili_izraz>
             <bin_xili_izraz>
              <bin_i_izraz>
               <jednakosni_izraz>
                <odnosni_izraz>
                 <aditivni_izraz>
                  <multiplikativni_izraz>
                   <cast_izraz>
                    <unarni_izraz>
                     OP_INC 7 ++
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        IDN 7 i
        D_ZAGRADA 7 )
        <naredba>
         <slozena_naredba>
          L_VIT_ZAGRADA 7 {
          <lista_naredbi>
           <naredba>
            <izraz_naredba>
             <izraz>
              <izraz_pridruzivanja>
               <postfiks_izraz>
                <primarni_izraz>
                 IDN 8 y
               OP_PRIDRUZI 8 =
               <izraz_pridruzivanja>
                <log_ili_izraz>
                 <log_i_izraz>
                  <bin_ili_izraz>
                   <bin_xili_izraz>
                    <bin_i_izraz>
                     <jednakosni_izraz>
                      <odnosni_izraz>
                       <aditivni_izraz>
                        <multiplikativni_izraz>
                         <cast_izraz>
                          <unarni_izraz>
                           <postfiks_izraz>
                            <primarni_izraz>
                             BROJ 8 1
             TOCKAZAREZ 8 ;
          D_VIT_ZAGRADA 9 }
     <naredba>
      <naredba_skoka>
       KR_RETURN 10 return
       <izraz>
        <izraz_pridruzivanja>
         <log_ili_izraz>
          <log_i_izraz>
           <bin_ili_izraz>
            <bin_xili_izraz>
             <bin_i_izraz>
              <jednakosni_izraz>
               <odnosni_izraz>
                <aditivni_izraz>
                 <multiplikativni_izraz>
                  <cast_izraz>
                   <unarni_izraz>
                    <postfiks_izraz>
                     <primarni_izraz>
                      IDN 10 y
       TOCKAZAREZ 10 ;
    D_VIT_ZAGRADA 11 }
