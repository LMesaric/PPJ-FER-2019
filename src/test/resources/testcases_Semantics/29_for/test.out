<naredba_petlje> ::= KR_FOR(7,for) L_ZAGRADA(7,() <izraz_naredba> <izraz_naredba> <izraz> D_ZAGRADA(7,)) <naredba>
